cli/docker_handler.go

```diff
@@ -86,18 +87,49 @@ func (c *DockerHandler) ConfigFromLabelsEnabled() bool {
 	return c.configsFromLabels
 }
 
+// Watch for Docker events and update the labels accordingly
 func (c *DockerHandler) watch() {
-	const pollInterval = 10 * time.Second
-	c.logger.Debugf("Watching for Docker labels changes every %s...", pollInterval)
-	ticker := time.NewTicker(pollInterval)
-	defer ticker.Stop()
-	for range ticker.C {
-		labels, err := c.GetDockerLabels()
-		// Do not print or care if there is no container up right now
-		if err != nil && !errors.Is(err, errNoContainersMatchingFilters) {
-			c.logger.Debugf("%v", err)
+	c.logger.Debugf("Listening for Docker events to hot reload configs...")
+
+	events := make(chan *docker.APIEvents)
+	err := c.dockerClient.AddEventListener(events)
+	if err != nil {
+		c.logger.Errorf("Error adding event listener: %v", err)
+	}
+	defer func() {
+		// remove the listener when the program exits
+		err := c.dockerClient.RemoveEventListener(events)
+		if err != nil {
+			c.logger.Errorf("Error removing event listener: %v", err)
+		}
+	}()
+
+	lifecycleEvents := []string{
+		"create",
+		"start",
+		"restart",
+		"stop",
+		"kill",
+		"die",
+		"destroy",
+	}
+
+	otherManagementEvents := []string{
+		"pause",
+		"unpause",
+		"rename",
+		"update",
+	}
+
+	for event := range events {
+		if event.Type == "container" && (slices.Contains(lifecycleEvents, event.Action) || slices.Contains(otherManagementEvents, event.Action)) {
```

Comment:
Because supported event types are defined in slices `slice.Contains` will have to iterate over those two slices on every received event. It would be much more efficient to store them as switch/case or a map[^1]

Something like this will work better:

```suggestion

         shouldProcessEvent := func(event *docker.APIEvents) bool {
             if event.Type != "container":
                return false
             }
         
             switch (event.Action) {
              // lifecycle events
              case "create", "start", "restart", "stop", "kill", "die", "destroy": 
                  return true
              // other management events
              case "pause", "unpause", "rename", "update":
                  return true
              default:
                 return false
             }
         }

	for event := range events {
		if !shouldProcessEvent {
		    continue
		}
```


[^1]: https://adayinthelifeof.nl/2021/03/04/go-map-vs-switch.html